sdks/python/apache_beam/metrics/execution.py

```diff
@@ -135,18 +136,40 @@ def set_metrics_supported(self, supported):
     with self._METRICS_SUPPORTED_LOCK:
       self.METRICS_SUPPORTED = supported
 
-  def current_container(self):
+  def _old_style_container(self):
+    """Gets the current MetricsContainer based on the container stack.
+
+    The container stack is the old method, and will be deprecated. Should
+    rely on StateSampler instead."""
     self.set_container_stack()
     index = len(self.PER_THREAD.container) - 1
     if index < 0:
       return None
     return self.PER_THREAD.container[index]
 
+  def current_container(self):
+    """Returns the current MetricsContainer."""
+    sampler = statesampler.EXECUTION_STATE_SAMPLERS.current_sampler()
+    if sampler is None:
```

Comment:
When would this be false? Can we simply assert it'll be present after your CL rather than keeping both the old and new paths in parallel?